src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyNamingPolicy.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Determines the policy used to convert a JSON name to another format, such as a camel-casing format.
+    /// </summary>
+    public abstract class JsonPropertyNamingPolicy
+    {
+        protected JsonPropertyNamingPolicy() { }
+
+        /// <summary>
+        /// Returns the policy for camel-casing.
+        /// </summary>
+        public static JsonPropertyNamingPolicy CamelCase { get; } = new JsonCamelCaseNamePolicy();
+
+        /// <summary>
+        /// Converts the provided name.
+        /// </summary>
+        /// <param name="name">The name to convert.</param>
+        /// <returns>The converted name</returns>
+        protected abstract string ConvertName(string name);
```

Comment:
I think this should be public. When people create their own `JsonConverter` types (I forget what it is called in S.T.J) then they could want to call a policy to convert property names.

```cs
public void WriteJson(Dictionary<string, Account> value, Utf8JsonWriter writer)
{
    writer.WriteStartObject();
    foreach (var kvp in value)
    {
        // Camel case the key in the JSON output
        writer.WriteString(JsonPropertyNamingPolicy.CamelCase.ConvertName(kvp.Key), value.UserName);
    }
    writer.WriteEndObject();
}
```

Reply:
Example of me doing this in a converter for serializing `DataTable` to/from JSON: https://github.com/JamesNK/Newtonsoft.Json/blob/f80a8184c8c825a6ba63a22caa73b3235b3bdc49/Src/Newtonsoft.Json/Converters/DataTableConverter.cs#L73

Reply:
I did have a comment in the API issue about whether this should be made public or not. Due to no feedback, I kept it protected. However, I agree with your scenario above and the public status.

Reply:
Changing to public unless I hear otherwise.